routers/api/v1/repo/repo.go

```diff
@@ -178,7 +178,8 @@ func Search(ctx *context.APIContext) {
 		opts.IsPrivate = optional.Some(ctx.FormBool("is_private"))
 	}
 
-	sortMode := ctx.FormString("sort")
+	// as sort is an url param, we should be case insensitive
+	sortMode := strings.ToLower(ctx.FormString("sort"))
```

Comment:
```suggestion
	sortMode := ctx.FormString("sort")
```
Meanless

Reply:
please tell me why it is

Reply:
Why "as sort is an url param, it should be case insensitive"? What's the real use-case? Any other servers do so?

Why you (and the server) would like to pay the cost for end-developer's incorrect-case mistake?

Reply:
that's a cursed question ...
... in this case you could also argue why we do trim issue titles on creation ... it should be the users vault if the title contains spaces ...

Reply:
> ... in this case you could also argue why we do trim issue titles on creation

That's different. Title is inputted by end users. But here, these "sort modes" are "constant values".

If you considered the "sort mode" should be case-insensitive, then:

* you are inventing a new mechanism which never appeared in Gitea code base.
* you only hacky-patched this parameter in this function, but all (almost) other functions do not use case-insensitive constant values. And even in this function: `FormBool` doesn't accept `tRUE`, `sortOrder` doesn't use case-insensitive, the system is just breaking up to different mechanisms.

Reply:
well for the sace of consistence ...